BootloaderCommonPkg/Library/ElfLib/ElfLib.c

```diff
@@ -318,11 +318,14 @@ ParseElfImage (
       End = ALIGN_VALUE (SegInfo.MemAddr + SegInfo.MemLen, SegInfo.Alignment) - 1;
     }
   }
-  //
-  // 0 - MAX_UINT32  + 1 equals to 0.
-  //
-  ElfCt->ImageSize             = End - Base + 1;
-  ElfCt->PreferredImageAddress = (VOID *) Base;
+
+  if (End == 0 && Base == MAX_UINT32) {
```

Comment:
A better and more robust check:   if (End < Base || End == 0 || Base == MAX_UINT32) {